src/Sulu/Bundle/AdminBundle/Command/DownloadBuildCommand.php

```diff
@@ -0,0 +1,160 @@
+
+    /**
+     * @var string
+     */
+    private $remoteArchive;
+
+    const ASSETS_DIR = DIRECTORY_SEPARATOR . 'assets' . DIRECTORY_SEPARATOR . 'admin' . DIRECTORY_SEPARATOR;
+
+    const BUILD_DIR = DIRECTORY_SEPARATOR . 'public' . DIRECTORY_SEPARATOR . 'build' . DIRECTORY_SEPARATOR . 'admin';
+
+    const REPOSITORY_NAME = 'skeleton';
+
+    public function __construct(HttpClientInterface $httpClient, string $projectDir, string $suluVersion)
+    {
+        parent::__construct();
+
+        $this->httpClient = $httpClient;
+        $this->projectDir = $projectDir;
+        $this->suluVersion = $suluVersion;
+        $this->remoteRepository = 'https://raw.githubusercontent.com/sulu/skeleton/' . $suluVersion;
+        $this->remoteArchive = 'https://codeload.github.com/sulu/skeleton/zip/' . $suluVersion;
+    }
+
+    protected function configure()
+    {
+        $this->setDescription('Downloads the current admin application build from the sulu/skeleton repository.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $output->writeln('<info>Checking for changed files...</info>');
+
+        $indexJs = static::ASSETS_DIR . 'index.js';
+        $packageJson = static::ASSETS_DIR . 'package.json';
+        $webpackConfigJs = static::ASSETS_DIR . 'webpack.config.js';
+
+        $localIndexJsHash = $this->getLocaleFileHash($indexJs);
+        $localPackageJsonHash = $this->getLocaleFileHash($packageJson);
+        $localWebpackConfigJsHash = $this->getLocaleFileHash($webpackConfigJs);
+
+        $remoteIndexJsHash = $this->getRemoteFileHash($indexJs);
+        $remotePackageJsonHash = $this->getRemoteFileHash($packageJson);
+        $remoteWebpackConfigJsHash = $this->getRemoteFileHash($webpackConfigJs);
+
+        if ($localIndexJsHash !== $remoteIndexJsHash
+            || $localPackageJsonHash !== $remotePackageJsonHash
+            || $localWebpackConfigJsHash !== $remoteWebpackConfigJsHash
+        ) {
+            throw new \Exception(
+                sprintf(
+                    'The files in the local "%s" folder do not match the ones in the remote repository "%s".' . PHP_EOL
+                    . 'Either the build has been modified, which means it has to be done manually with NPM or the files'
+                    . ' in your repository are outdated and have to be copied from the remote repository.',
+                    static::ASSETS_DIR,
+                    $this->remoteRepository
+                )
+            );
+        }
+
+        $tempDirectory = sys_get_temp_dir() . DIRECTORY_SEPARATOR . static::REPOSITORY_NAME . uniqid(rand(), true);
+        $tempFileZip = $tempDirectory . '.zip';
+
+        $output->writeln('<info>Download remote repository...</info>');
+        $response = $this->httpClient->request('GET', $this->remoteArchive, [
+            'on_progress' => function(int $downloaded, int $size) {
+                /* echo $downloaded . '/' . $size . PHP_EOL; */
```

Comment:
?

Reply:
Actually I wanted to show a progress bar, but that does not seem to work correctly :confused: Both values only show 0, which is not correct for sure...